### ISSUE TITLE
lint on `libc::strlen(some_cstr.as_ptr())`

### ISSUE THREAD

**Issue by joshtriplett** [rust-lang/rust-clippy#7145]
### What it does

This lint would flag the use of `libc::strlen` on a `CString` or `CStr` value, and suggest calling `to_bytes().len()` instead.

### Categories (optional)

- Kind: `clippy::complexity`

*What is the advantage of the recommended code over the original code*

This avoids calling an unsafe `libc` function. Currently, it also avoids calculating the length; in the future, this may become equivalent to calling `strlen` at runtime, but it'll still avoid the need for an unsafe `libc` call.

### Drawbacks

None.

### Example

```rust
libc::strlen(name.as_ptr())
```

Could be written as:

```rust
name.to_bytes().len()
```

**Comment by joshtriplett** [comment]
Inspired by https://github.com/rust-lang/rust/pull/84589#discussion_r621498842

**Comment by mgacek8** [comment]
What about for `CString` suggesting `as_bytes().len()` instead of `to_bytes().len()` ? 
[as_bytes()](https://doc.rust-lang.org/std/ffi/struct.CString.html#method.as_bytes) shouldn't calculate the length in the future.

**Comment by joshtriplett** [comment]
@mgacek8 That seems reasonable, yes. The lint should apply to either `CStr` or `CString`; the suggested replacement can be different for the two.

**Comment by mgacek8** [comment]
I'll give it a shot.
@rustbot claim
